perf/src/packet.rs

```diff
@@ -137,6 +137,20 @@ where
         .deserialize_from(data)
 }
 
+pub fn deserialize_from_with_limit<R, T>(reader: R) -> bincode::Result<T>
+where
+    R: Read,
+    T: DeserializeOwned,
+{
+    // with_limit causes pre-allocation size to be limited
+    // to prevent against memory exhaustion attacks.
+    bincode::options()
+        .with_limit(PACKET_DATA_SIZE as u64)
+        .with_fixint_encoding()
+        .allow_trailing_bytes()
+        .deserialize_from(reader)
+}
```

Comment:
Need this because `limited_deserialize` only reads from `&[u8]`,
but shred code reads from a `Cursor`.